cluster/storage/src/main/java/com/linkedin/openhouse/cluster/storage/hdfs/HdfsStorageClient.java

```diff
@@ -48,4 +49,14 @@ public FileSystem getNativeClient() {
   public StorageType.Type getStorageType() {
     return HDFS_TYPE;
   }
+
+  @Override
+  public boolean pathExists(String path) {
+    try {
+      return fs.exists(new Path(path));
+    } catch (IOException e) {
```

Comment:
does fs.exists throw an IOException for non-existing paths? thats weird

Reply:
Looks like it returns true or false: https://github.com/apache/hadoop/blob/cd2cffe73f909a106ba47653acf525220f2665cf/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FileSystem.java#L1860

But could throw an IOException if getFileStatus() throws an IOException.

I am wondering if it could throw an IOException in scenarios like HDFS is unavailable but the file exists?